### ISSUE TITLE
categories and tags are not generated

### ISSUE THREAD

**Issue by codeandcloud** [cotes2020/jekyll-theme-chirpy#57]
I have converted my `blogger template` to `jekyll template`. On conversion, the layout of an example post will be like
    ---
    layout: post
    title: Binding html select using knockoutjs and asp.net mvc
    date: '2013-11-25T18:42:00.001+05:30'
    author: Naveen
    categories: 
    - asp.net mvc
    - javascript
    - knockout.js
    tags:
    - asp.net mvc
    - javascript
    - knockout.js
    modified_time: '2013-11-25T18:54:00.520+05:30'
    ---
URL: https://codeandcloud.github.io/

The site have these issues
1. Getting 404'ed on clicking a tag on https://codeandcloud.github.io/tabs/tags/
2. Categories doesn't get displayed at all

When I run `$ bash tools/init.sh`, terminal shows

    $ bash tools/init.sh
    [INFO] Success to update lastmod for 41 post(s).
    warning: LF will be replaced by CRLF in _data/updates.yml.
    The file will have its original line endings in your working directory
    [master ec06b7d] [Automation] Updated the Lastmod for post(s).
    1 file changed, 41 insertions(+), 41 deletions(-)
    Updated the Lastmod for post(s).
Its missing 

    [INFO] Succeed! 3 category-pages created.
    [INFO] Succeed! 4 tag-pages created.

Why are those pages not created? What am I doing wrong?

Thanks for this wonderful theme.

**Comment by cotes2020** [comment]
Due to technical limitations, the scripting tool cannot read the multiline yaml array:
```
---
categories:
- asp.net mvc
- ....

tags:
- asp.net mvc
- ...
---
```
Change to single line will be fine:

```
---
categories: [item1, item2]      # notice that there are no more than two items.
tags: [item1, item2, ... itemN ]
---
```

**Comment by codeandcloud** [comment]
Ok. So the changes are 

1. make both array like
2. categories max two

I have an additional question. Will spaces in between work for tags/categories `server-side paging` ?

**Comment by cotes2020** [comment]
Spaces are allowed.

**Comment by codeandcloud** [comment]
Now the tag and category pages are generated. 

That part is done but the two errors mentioned [here](https://github.com/cotes2020/jekyll-theme-chirpy/issues/57#issue-622679696) persists.

Two examples of generated tag and category page are as follows

tag: `jquery`

![image](https://user-images.githubusercontent.com/872421/82602593-f4bfcc00-9bce-11ea-9561-f2eb55c17aa6.png)

category: `javascript`
![image](https://user-images.githubusercontent.com/872421/82602682-1751e500-9bcf-11ea-922a-89789d501b95.png)

Is this because **the files inside the _posts folder are .html files** and not `.md`?

Does a category should necessarily have two sub-categories?
I have either 1 or 2 categories in all pages. 

Is there a problem if I omit categories altogether?
Is there a condition that category name and tag name cannot be same?

**Comment by cotes2020** [comment]
Yes, the page generator script only read the `*.md` files. And the details of categories and tags, please read [this post](https://chirpy.cotes.info/posts/write-a-new-post/) of demo site.

**Comment by codeandcloud** [comment]
Thanks. It was due to a error while `jekyll serve`

I found two minor things

1. Suppose I use a sub-category "General" for category "ASP.NET" and "JavaScript" categories. The posts get aligned wrong. (I know its because the front matter will be same for same category)
2. If I have a category list [JavaScript, jQuery] for a post, adding category list [JavaScript] to another post fails with error in line 18 of `tabs/categories.md`

Just letting you know. I guess you already knew it and its by design.

Thanks for the wonderful design. Looks Minimalist and chic. My style. Loved it
And thanks also for the prompt response. Much appreciated. <3

Closing this issue as it's resolved

**Comment by cotes2020** [comment]
>1. Suppose I use a sub-category "General" for category "ASP.NET" and "JavaScript" categories. The posts get aligned wrong. (I know its because the front matter will be same for same category)

Indeed, the name and hierarchy of each `category` must be unique throughout the site.

>2. If I have a category list [JavaScript, jQuery] for a post, adding category list [JavaScript] to another post fails with error in line 18 of tabs/categories.md

Thank you for your clue! There was a similar report on #48, but I didn't notice it was a bug at that time. It has now been fixed at 451ede05dd948e460c83f76c35bc9fd0679619e6, please merge it into your branch.

**Comment by codeandcloud** [comment]
Thanks @cotes2020. I was inventing second category for most posts :)
